Core/src/TrackFinding/MeasurementSelector.cpp

```diff
@@ -15,17 +15,41 @@
 #include "Acts/EventData/Types.hpp"
 
 #include <algorithm>
+#include <limits>
 
 namespace Acts {
 
 MeasurementSelector::MeasurementSelector()
-    : m_config{{GeometryIdentifier(), MeasurementSelectorCuts{}}} {}
+    : MeasurementSelector({{GeometryIdentifier(), MeasurementSelectorCuts{}}}) {
+}
 
 MeasurementSelector::MeasurementSelector(const MeasurementSelectorCuts& cuts)
-    : m_config{{GeometryIdentifier(), cuts}} {}
+    : MeasurementSelector({{GeometryIdentifier(), cuts}}) {}
 
 MeasurementSelector::MeasurementSelector(Config config)
-    : m_config(std::move(config)) {}
+    : m_config(std::move(config)) {
+  for (const auto& cuts : m_config) {
+    validateCuts(cuts);
+  }
+}
+
+void MeasurementSelector::validateCuts(const MeasurementSelectorCuts& cuts) {
+  if (!cuts.chi2CutOffOutlier.empty() &&
+      cuts.chi2CutOff.size() != cuts.chi2CutOffOutlier.size()) {
+    throw std::invalid_argument(
+        "chi2CutOff and chi2CutOffOutlier must have the same size");
+  }
+
+  if (cuts.chi2CutOff.size() != cuts.numMeasurementsCutOff.size()) {
+    throw std::invalid_argument(
+        "chi2CutOff and numMeasurementsCutOff must have the same size");
+  }
+
+  if (cuts.chi2CutOff.size() != cuts.etaBins.size() + 1) {
+    throw std::invalid_argument(
+        "chi2CutOff must have one more element than etaBins");
+  }
```

Comment:
I guess we no longer want all these checks: just that `chi2CutOff` and `numMeasurementsCutOff` aren't empty, since those aren't allowed to be empty in `getCutsByTheta`.